### PR TITLE
signup and login function

### PR DESCRIPTION
**Pull request by Miraclepio**

### CODE REVIEW

**Review by Lu-Tola07 — APPROVED** [review]
You were to do the SignUp function alone, but thank you.

**Comment by Miraclepio** [comment]
good

File: controller/userController.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      process.env.JWT_SECRET, 
+      { expiresIn: "3h" } 
+    );
+
+    res.status(200).json({ message: "Login successful", token: userToken });
+  } catch (error) {
+    console.error("Error logging in:", error);
+    res.status(500).json({ message: "Internal Server Error" });
+  }
+};
```
Comment (folakeatolagbe96) [inline controller/userController.js:109]: nice work but you can do better and try adding commenting for other reading through your code

**Review by folakeatolagbe96 — APPROVED** [review]

File: controller/userController.js (hunk 1 above)
Comment (Miraclepio) [reply controller/userController.js:109]: certainly!

**Review by Sylvergold — APPROVED** [review]
Approve

**Review by Cjoseph8 — APPROVED** [review]
nicely  done

**Pull request merged**